### PR TITLE
Creating/Updating Util files

### PR DESCRIPTION
**Pull request by ChaseTiong**
# New / Updated Files
> * Creation of FilterUtil to be used for clear, find and list controller , in order to refactor the code into linear style.
> * Updating of StringUtil changing method naming
> * Updating of DateUtil extract out parseMethods to be used by controllers
> * Creation of ParseUtil to parse result from tokenizer which will be used by controllers

# New / Updated test cases
> * Update it with new test cases for the methods I have created

### CODE REVIEW

File: src/main/java/seedu/todo/commons/util/DateUtil.java [hunk 1]
```diff
@@ -197,5 +218,80 @@ public static LocalDateTime parseDateTime(String dateTimeString) {
...
+     */
+    public static boolean checkIfDateExist(LocalDateTime date) {
+        LocalDateTime currentDate = LocalDateTime.now();
+        return currentDate.getDayOfYear() != date.getDayOfYear() || currentDate.getMonth() != date.getMonth() || 
+                currentDate.getYear() != date.getYear();
+    }
+    
+    /*
+     * To be used to parse natural date into LocalDateTime 
+     * @parser Natty
```
Comment (louietyj) [inline src/main/java/seedu/todo/commons/util/DateUtil.java]: What the hell is this line, I've never seen such code documentation markup before.

File: src/main/java/seedu/todo/commons/util/DateUtil.java [hunk 2]
```diff
@@ -197,5 +218,80 @@ public static LocalDateTime parseDateTime(String dateTimeString) {
...
+     *
```
Comment (louietyj) [inline src/main/java/seedu/todo/commons/util/DateUtil.java]: Please remove this method, it has been extracted into DateParser. And you're claiming credit for my method.

**Review by louietyj — DISMISSED** [review]

File: src/main/java/seedu/todo/commons/util/DateUtil.java (hunk 2 above)
Comment (ChaseTiong) [reply src/main/java/seedu/todo/commons/util/DateUtil.java]: First of all, I have extracted out this method like on Monday/Tuesday. 
Secondly, I didn't put my matric number as it is created by you so I am not claiming any credit for your methods.
Will remove it as suggested.

Comment (ChaseTiong) [reply src/main/java/seedu/todo/commons/util/DateUtil.java]: removed

**Comment by louietyj** [comment]
Only open a PR when you want your code to be reviewed and merged, not when you are still actively working on it.

**Comment by ChaseTiong** [comment]
@louietyj Sorry. I am done with it already. Thanks for notifying me

File: src/main/java/seedu/todo/commons/util/ParseUtil.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
+package seedu.todo.commons.util;
+
+import java.util.Map;
+
+/**
+ * Helper functions for parsing.
+ * 
+ * @@author A0139922Y
+ */
+public class ParseUtil {
```
Comment (irvinlim) [inline src/main/java/seedu/todo/commons/util/ParseUtil.java]: Can I recommend that this goes inside of controllers/concerns/DateParser.java? Merge master into your branch and move it there.

**Comment by louietyj** [comment]
@irvinlim Code-wise, LGTM. But this seems to be just a bunch of utility methods (with unit tests) that aren't used anywhere, functionality-wise. We will have to scrutinize much harder when a PR comes in that makes use of these utility methods.

@ChaseYaoCong This may be a bit debatable, but you generally should not be separating PRs by modules / files. PRs should be contain logical units of work that solve an issue. Instead, it's better to incrementally create PRs that add util methods together with the functionality that actually uses the util. Otherwise, we have no idea what the util methods are actually used for, and when the PR that uses it comes in, we will have to come back and think about whether these util methods even make sense in the context of what they are used for.

**Review by irvinlim — CHANGES_REQUESTED** [review]
@louietyj I think ParseUtil should be combined with DateParser. Otherwise its ok.

@ChaseYaoCong Additionally, I'm updating the authorship for some methods, it's better to put authorship for every single method because of how the collate thing works, wait for my commit

File: src/main/java/seedu/todo/commons/util/ParseUtil.java (hunk 3 above)
Comment (ChaseTiong) [reply src/main/java/seedu/todo/commons/util/ParseUtil.java]: @irvinlim okay. 
@louietyj do you want me to combine refactored clear controller with this PR? So I can also fixes #146 at the same time and use the Util methods I created

**Pull request closed without merging**